src/v/cluster_link/model/types.h

```diff
@@ -55,6 +55,38 @@ static constexpr std::string_view to_string_view(mirror_topic_state s) {
     return "unknown";
 }
 
+enum class task_state : uint8_t {
+    /// The task is currently active and processing
+    active,
+    /// The task has been paused
+    paused,
+    /// The task has critically failed and will not resume
+    task_failed,
+    /// The link to the source cluster is unavailable.  This could be a
+    /// temporary condition that can be resolved by changing configuration of
+    /// the task or on the source cluster.  This state is informative and the
+    /// task will continue to run at its set interval
+    link_unavailable,
+    /// The task is not configured to run
+    not_running,
+};
+
+static constexpr std::string_view to_string_view(task_state st) {
+    switch (st) {
+    case task_state::active:
+        return "active";
+    case task_state::paused:
+        return "paused";
+    case task_state::task_failed:
+        return "task_failed";
+    case task_state::link_unavailable:
+        return "link_unavailable";
+    case task_state::not_running:
+        return "not_running";
+    }
+    return "unknown";
```

Comment:
i think if the expectation is that all the cases are covered, then use `assert(false` or `unreachable()`. if it's not fully covered, then use `default:`.